Sources/IdentityState.swift

```diff
@@ -175,6 +206,46 @@ class IdentityState {
         return true
     }
 
+    /// Determines if we should update the advertising identifier with `newAdID` and if the advertising tracking consent has changed.
+    /// - Parameter newAdID: the new ad id
+    /// - Returns: A tuple indicating if the ad id has changed, and if the consent should be updated
+    private func shouldUpdateAdId(newAdID: String?) -> (adIdChanged: Bool, updateConsent: Bool) {
+        guard let newAdID = newAdID else { return (false, false) }
+
+        let existingAdId = identityProperties.advertisingIdentifier ?? ""
+
+        // did the advertising identifier change?
+        if (!newAdID.isEmpty && newAdID != existingAdId)
+            || (newAdID.isEmpty && !existingAdId.isEmpty) {
+            // Now we know the value changed, but did it change to/from null?
+            // Handle case where existingAdId loaded from persistence with all zeros and new value is not empty.
```

Comment:
Do we need to handle this case here since we never stored an all-zeroes value?

Reply:
Sorry, I'm not sure I understand; is an all-zeros value case reaching this point impossible in the current flow?

Reply:
We had this use-case in the Identity non-Edge workflow since we were previously also storing all-zeroes values for advertising id. But the Identity Edge extension never handled advertising id and all-zero values are considered "" in your implementation. Thoughts? 
cc @kevinlind

Reply:
Thanks for the background info, that's helpful in understanding the overall flow. I checked the `set` flow for new `advertisingIdentifier` values and I couldn't find any logic explicitly filtering out the all-zero case and converting it to the empty string? 

If I'm understanding correctly, the update flow is:
event passed by listener to -> 
`handleRequestContent(event:)` ->
`state.updateAdvertisingIdentifier(event:createXDMSharedState:eventDispatcher:)` ->
`shouldUpdateAdId(newAdID:)` (where the main logic for what should happen with incoming AdIDs happens, but no value mutation/modification) ->
`IdentityProperties.advertisingIdentifier` (setter) 
```swift
guard let newAdId = newValue, !newAdId.isEmpty else {
    return // new ID is nil or empty
}
```
The setter for `advertisingIdentifier` doesn't modify/convert any values, only filters out nil and empty string values. So I was thinking the all-zeros case should be able to be set as a valid value in `advertisingIdentifier`.

I do have a question based on the current flow and the spec from the wiki "Advertising ID handling in Edge Identity":
Currently the `shouldUpdateAdId(newAdID:)` logic immediately returns false for adIdChanged if the incoming value is nil, but isn't nil a valid case for removing the currently set AdID (as defined in the spec cases, scenario 1)? The main consequence of the current logic is that sending `nil` will not trigger the consent change flow for scenario 1
```swift
private func shouldUpdateAdId(newAdID: String?) -> (adIdChanged: Bool, updateConsent: Bool) {
        guard let newAdID = newAdID else { return (false, false) }
....
```

Reply:
Emilia has a valid point. As this is new functionality and  `Event + Identity` converts an advertising ID of all zeros to an empty string, there isn't a way to set an advertising ID of all zeros. Also, it appears there is no migration path to Edge Identity, so there will not be any persisted values from v5 migrated to Edge Identity. I think it's safe to remove the  `existingAdId == IdentityConstants.Default.ZERO_ADVERTISING_ID` check below. So we can just do
```
if newAdID.isEmpty || existingAdId.isEmpty  {
                return (true, true)
            }
```